keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/TensorMulByMatrixMul.java

```diff
@@ -0,0 +1,86 @@
+package io.improbable.keanu.tensor.dbl;
+
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TensorMulByMatrixMul {
+
+    public static DoubleTensor tensorMmul(DoubleTensor a, DoubleTensor b, int[][] axes) {
```

Comment:
This is a bit of a whopper. Can we get some java doc or break it out into methods? I've got no idea what's going on to be honest

Reply:
This came from ND4J mostly. I can break it out and leave the proper credit to ND4J at the top in a class comment.